test/test-clusters/src/main/java/org/elasticsearch/test/cluster/local/LocalClusterFactory.java

```diff
@@ -407,6 +412,21 @@ private void addKeystoreFiles() {
             });
         }
 
+        private void writeSecureSecretsFile() {
+            if (spec.getSecrets().isEmpty() == false) {
+                try {
+                    Path secretsFile = configDir.resolve("secrets/secrets.json");
```

Comment:
This will work in serverless, but not in the default distribution. The file based secrets support is only added there.

Reply:
That's fine. This exists only for when file-based secure settings is enabled. There's a whole separate DSL for configuring secrets via this method that will only be used when appropriate.